### ISSUE TITLE
Incorrect behaviours when including external entities from .txt

### ISSUE THREAD

**Issue by BalduinLandolt** [eclipse/lemminx#816]
I have enoucntered one bug and a couple of (for me) unexpected behaviours when including external entities from a .txt file.  
Consider the following example:

```XML
<?xml version="1.0" encoding="UTF-8"?>

<!DOCTYPE root 
[
<!ENTITY % Menota_entities SYSTEM 'http://www.menota.org/menota-entities.txt'>
%Menota_entities; ]>

<root>
    <text>
        sample text with entity: &sol; &emacr;
    </text>
</root>
```

This includes an entity from from an external .txt file that consists of several entity declarations.  
(like `<!-- LATIN SMALL LETTER E WITH MACRON -->		<!ENTITY emacr "&#x0113;">`)

The so defined entities clearly work, as they don't produce an error (see screenshot below). However, vs code does not behave as it should, in this context:  
(I'll compare it to the behaviour in Oxygen XML just for reference.)
- there is a code completion for `&Menota_entities;` which results in an error. (See screenshot; in Oxygen, I don't get this completion suggestion, when I add it however, I get the same error.)
- there are no code completions for the actual entities like `&sol;`, `&emacr;` etc. (In Oxygen, I get these code completions)
- I also don't get a code lense popup for these entites (what it stands for, where it's declared, etc.)

![grafik](https://user-images.githubusercontent.com/33053745/86528809-dc202280-beab-11ea-9491-7a800613dc63.png)

---

When describing this problem I came across another thing (which might be a separate issue, though):

When I declare an entity like this: `<!-- random entity--> <!ENTITY blah "blaaaaaah">`, I would expect code lense to display the comment ("random entity"), when I hover over `%blah;`. (At least that's how Oxygen XML does it, and i find that quite useful.)

**Comment by angelozerr** [comment]
> there is a code completion for &Menota_entities; which results in an error. (See screenshot; in Oxygen, I don't get this completion suggestion, when I add it however, I get the same error.)

Indeed it's a bug that we must fix. Need investigation to know if it's easy to fix it?

> there are no code completions for the actual entities like &sol;, &emacr; etc. (In Oxygen, I get these code completions)

Indeed it's a bug that we must fix. Need investigation to know if it's easy to fix it?

> I also don't get a code lense popup for these entites (what it stands for, where it's declared, etc.)

What do you mean with codelens? You would like to have codelens for declared entites with number of references?

**Comment by BalduinLandolt** [comment]
> What do you mean with codelens? You would like to have codelens for declared entites with number of references?

No, sorry, I didn't mean code lens. I meant the popup that VS Code has (see image below) when I hover over an entity. Entities declared in the doctype have a popup, entities from the .txt don't.
![grafik](https://user-images.githubusercontent.com/33053745/86540167-5fb92e00-bf03-11ea-8903-1630a7cddb00.png)
I'm honestly not sure if that is done by the language server or by the vs code extension. I simply assumed it might be cuased by the same problem that causes there not to be code completion for them.
